File: jetbrains-plugin/src/main/kotlin/com/ezevillo/gitreview/domain/FinishOutcome.kt
package com.ezevillo.gitreview.domain

enum class FinishOutcome {
    NO_EDITS,
    PENDING,
}

/**
 * Decide toast after successful `finish` from refreshed state only — never
 * parse finish human stdout.
 */
fun finishOutcome(refreshed: ReviewState, branch: String): FinishOutcome {
    val pending = refreshed.branches.any { it.name == branch && it.finish?.state == "pending" }
    return if (pending) FinishOutcome.PENDING else FinishOutcome.NO_EDITS
}

File: visualstudio-extension/src/GitReview.Domain/FinishOutcome.cs
namespace GitReview.Domain;

public enum FinishOutcome
{
    NoEdits,
    Pending,
}

public static class FinishOutcomeLogic
{
    /// <summary>
    /// Decide toast after successful finish from refreshed state only — never parse finish human stdout.
    /// </summary>
    public static FinishOutcome FinishOutcome(ReviewState refreshed, string branch)
    {
        var pending = refreshed.BranchesList.Any(b => b.Name == branch && b.Finish?.State == "pending");
        return pending ? Domain.FinishOutcome.Pending : Domain.FinishOutcome.NoEdits;
    }
}
